TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/ExpressionBasedNodeLabelGenerator.php

```diff
@@ -71,13 +74,27 @@ public function initializeObject()
      */
     public function getLabel(NodeInterface $node, $crop = true)
     {
-        $label = \TYPO3\Eel\Utility::evaluateEelExpression($this->getExpression(), $this->eelEvaluator, array('node' => $node), $this->defaultContextConfiguration);
+        $label = null;
+        if ($node->getNodeType()->isOfType('TYPO3.Neos:ContentCollection') && $node->isAutoCreated()) {
+            $parentNode = $node->getParent();
+            $property = 'childNodes.' . $node->getName() . '.label';
+            if ($parentNode->getNodeType()->hasConfiguration($property)) {
+                $expression = $parentNode->getNodeType()->getConfiguration($property);
```

Comment:
I'd prefer if this also accepted simple strings and not only expressions, since that's unnecessary most of the time